cmake/extensions.cmake

```diff
@@ -96,6 +96,7 @@ function(add_overlay image overlay_file overlay_type)
     set(${image}_${overlay_type} "${old_overlays};${overlay_file}" CACHE STRING
       "Extra config fragments for ${image} child image" FORCE
     )
+    set(${image}_${overlay_type} "${old_overlays};${overlay_file}" PARENT_SCOPE)
```

Comment:
this is just setting the overlay in parent scope, that is the scope of the caller, and therefore not really fixing anything, cause there could be higher level scopes.

But besides that, if line:93 is hiding the cache variable then we have a different issue cause for child images then those variables should never be in local scope, they should only exists in the cache.

When does this occur  / which sample ?
And is it a problem related to when this function is called, for example:
- add_child_image(NAME my_image ...)
- add_overlay(my_image ...)  
cause in such a case the overlay file is set too late and things will not work.

Reply:
found the root cause.